### ISSUE TITLE
Huge S3 data leak after letting Azuracast connected to remote aws storage

### ISSUE THREAD

**Issue by ImaCrea** [AzuraCast/AzuraCast#3633]
Hi there,
short story : our aws S3 bill exploded for last December. Unfortunately I only discover today, hundreds of euros lost 😣.

I'm investigating but what it seems to be linked with Azuracast being connected to our s3 storage because it started on the date I connected them together. I couldn't make it work properly (see #3551 ) and I left idly working. I wasn't even really streaming to anybody and on the player seemed to play only the demo jingle on loop.

But something else was happening in the background. Look at this bill :
![Capture d’écran 2021-01-04 à 23 07 46](https://user-images.githubusercontent.com/764332/103584474-aeab4480-4ee1-11eb-84e0-9cb0ac8cdc84.png)

I killed the connection, deleted the iam profile and even shutdown the server and it seems we're now back to normal. So it looks like its very connected.

I don't understand what happened and I don't see why there's been so much traffic going on. What's the logic implemented behind this remote storage connection?

It took me all this time to realize because we're used to have a very average cost which has been the same for a few years now so I only look for a detail at the end of the month :/

Any idea about what happened and how to prevent such thing?

**Comment by SC2Mitch** [comment]
I'll investigate and see why it's pulling so much power, I don't think this is intended

**Comment by SlvrEagle23** [comment]
@ImaCrea I looked at several stations after the S3/Dropbox rollout to see what was causing them particular problems, and one of the things I found was that AzuraCast was repeatedly trying to reprocess files that it couldn't process, not just a few times a day or a week but every 5 minutes every day. In the case of local filesystems, this isn't an issue at all other than the unnecessary consumption of a little bit of CPU; on the remote systems like S3/Dropbox, this means we pull the file to analyze it locally, and that is likely the cause of the fees you're running into.

I would suggest two things to help immediately mitigate this issue for you:

 - Update to the latest Rolling Release channel version (at least 2021-01-01 or newer) to take advantage of the new database that tracks unprocessable files and won't reprocess them again for several days (or if it sees a change in their modified timestamp). This should significantly reduce the egress traffic you're seeing from S3/Dropbox. Regardless of anything we do, there are still some fees involved in downloading the files when instructing Liquidsoap to play them, but this should help a lot with the processing files issue.
 - Consider switching to an S3-compatible provider that doesn't charge on egress, like Wasabi. Using S3 isn't a problem for AzuraCast but as mentioned in the above bullet point, you'll always have _some_ relatively constant amount of egress traffic, and there's no need to pay for it when there are competing services that don't charge you for it. Without being hit with egress traffic bills, even at its worst performance AzuraCast would be an identical bill to your regular bill.
